ipatests/test_integration/test_commands.py

```diff
@@ -1235,6 +1235,22 @@ def test_cb(stdout):
         cmd = ["journalctl", "-u", "sshd", f"--since={since}"]
         tasks.run_repeatedly(self.master, command=cmd, test=test_cb)
 
+    def test_user_del_preserve(self):
+        """
+        Test if preserving a user account produces correct output message.
+
+        https://pagure.io/freeipa/issue/9187
+        """
+        user = "testuser"
+        tasks.kinit_admin(self.master)
+        tasks.create_active_user(self.master, user, "Secret123")
+
+        result = tasks.user_del(self.master, user, preserve=True)
```

Comment:
since `create_active_user` `kinit`s the newly created user, it's necessary to purge the cache and re-`kinit` the admin. 

In this case, it would be
```
tasks.kdestroy_all(self.msater)
tasks.kinit_admin(self.master)
```